### PR TITLE
eth/protocols, metrics, p2p: add handler performance metrics

### PR DESCRIPTION
**Pull request by karalabe**
Currently we have no insight into how much time serving specific network requests take. This is a problem because we can't correlate performance issues with network traffic. This PR adds a measurement to all `eth` and `snap` packet handlers to track the latency of handling each packet type in a histogram. Hopefully this will shed some light onto how much time we actually take to serve certain requests and what might influence it.

TL;DR: PR makes colored pixels

Screenshot from 2021-03-26 14-03-58

### CODE REVIEW

**Review by holiman — APPROVED** [review]
colored pixels guud
